src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -336,22 +336,42 @@ public async Task BulkUpdateSearchParameterIndicesAsync(IReadOnlyCollection<Reso
                     sqlCommandWrapper,
                     _bulkReindexResourcesTvpGeneratorVLatest.Generate(resources.ToList()));
 
-                try
+                if (_schemaInformation.Current >= SchemaVersionConstants.BulkReindexReturnsFailuresVersion)
                 {
-                    await sqlCommandWrapper.ExecuteNonQueryAsync(cancellationToken);
+                    // We will reindex the rest of the batch if one resource has a versioning conflict
+                    var failedResourceCount = (int?)await sqlCommandWrapper.ExecuteScalarAsync(cancellationToken);
+
+                    if (failedResourceCount != 0)
+                    {
+                        string message = string.Format(Core.Resources.ReindexingResourceVersionConflictWithCount, failedResourceCount);
+                        string userAction = string.Format(Core.Resources.ReindexingUserAction);
+
+                        _logger.LogError(message);
+                        throw new PreconditionFailedException(message + " " + userAction);
+                    }
                 }
-                catch (SqlException e)
+                else
                 {
-                    switch (e.Number)
+                    try
                     {
-                        // TODO: we should attempt to reindex resources that failed to be reindexed
-                        case SqlErrorCodes.PreconditionFailed:
-                            throw new PreconditionFailedException(string.Format(Core.Resources.ReindexingResourceVersionConflict));
-                        case SqlErrorCodes.NotFound:
-                            throw new ResourceNotFoundException(string.Format(Core.Resources.ReindexingResourceNotFound));
-                        default:
-                            _logger.LogError(e, "Error from SQL database on reindex");
-                            throw;
+                        // We are running an earlier schema version where we will fail the whole batch if there is a conflict
+                        await sqlCommandWrapper.ExecuteNonQueryAsync(cancellationToken);
+                    }
+                    catch (SqlException e)
+                    {
+                        switch (e.Number)
+                        {
+                            case SqlErrorCodes.PreconditionFailed:
+                                string message = string.Format(Core.Resources.ReindexingResourceVersionConflict);
+                                string userAction = string.Format(Core.Resources.ReindexingUserAction);
+
+                                _logger.LogError(message);
+                                throw new PreconditionFailedException(message + " " + userAction);
+
+                            default:
+                                _logger.LogError(e, string.Format(Core.Resources.ReindexingSqlError));
```

Comment:
I don't get reason behind `string.Format(ResourceString)`

`string.Format(str) == str`

Reply:
🤦🏻‍♀️ thanks, removed redundant calls to `string.Format()` in 57a3a31